src/hotspot/share/gc/g1/g1CollectedHeap.inline.hpp

```diff
@@ -313,4 +313,68 @@ inline bool G1CollectedHeap::is_collection_set_candidate(const HeapRegion* r) co
   return candidates->contains(r);
 }
 
+#define heap_locking_asserts_params(_extra_message_)                          \
+  "%s : Heap_lock locked: %s, at safepoint: %s, is VM thread: %s",            \
+  (_extra_message_),                                                          \
+  BOOL_TO_STR(Heap_lock->owned_by_self()),                                    \
+  BOOL_TO_STR(SafepointSynchronize::is_at_safepoint()),                       \
+  BOOL_TO_STR(Thread::current()->is_VM_thread())
+
+DEBUG_ONLY(
```

Comment:
We generally only use DEBUG_ONLY style macros for single small expressions, not for large multi-line
blocks of code.  For those we use `#ifdef ASSERT` and the like.